README.md

```diff
@@ -77,19 +97,33 @@ The naming scheme used for the generated files is controlled by the `namingSchem
 * Spaces are removed, camel-casing the title.
 * If the feature file starts with a digit, the classname is prefixed with '_'
 * A on up counter is appended to the classname, to prevent clashes. |
+| pattern       | Generate the filename based on the `namingPattern` property.  
+The following tokens can be used in the pattern:
+* `{f}` Converts the feature file name to a valid class name using the rules for feature-title, apart from the one up counter.
+* `{c}` Adds a one up counter. |
+
+By default, generated test files use the `simple` naming strategy.
+
+####Note on Pattern Naming Scheme
+The `pattern` naming scheme is for advanced usage only.  
 
-By default, `simple` naming strategy is used.
+It is up to you to ensure that class names generated are valid and there are no clashes.  If the same class name is generated multiple times, then it shall be overwritten and some of your tests will not be executed.
 
-By default, generated test files use a 'simple' naming scheme.  
+The `namingPattern` property is for the **class name** only.  Do not add the `.java` suffix.
 
 
```

Comment:
I think the Rerun functionality deserves a separate section heading with a bit more description:
* Why it's bad
* What it does
* How to enable it